### PR TITLE
Fix:Docs custom app example

### LINKED ISSUE #2302: Custom App documentation example - formMiddleware
**What you were expecting:**

By following tutorial at https://marmelab.com/react-admin/CustomApp.html I should be able to configure custom provider.

**What happened instead:**

Got an exception:
```js
redux.js:567 Uncaught TypeError: middleware is not a function
    at redux.js:567
    at Array.map (<anonymous>)
    at redux.js:566
    at <anonymous>:1:28482
    at Module.createStore (redux.js:87)
    at Object.configureStore [as default] (configure-store.ts:49)
    at Object../src/redux.ts (redux.ts:9)
    at __webpack_require__ (bootstrap:721)
    at fn (bootstrap:98)
    at Object../src/index.tsx (index.tsx:7)
    at __webpack_require__ (bootstrap:721)
    at fn (bootstrap:98)
    at Object.0 (serializer.ts:4)
    at __webpack_require__ (bootstrap:721)
    at ./node_modules/@babel/runtime/helpers/arrayWithoutHoles.js.module.exports (bootstrap:788)
    at bootstrap:788
```

**Steps to reproduce:**

Following the docs for CustomApp.

**Related code:**

https://github.com/marmelab/react-admin/pull/2249/files#diff-48a1c77367d4a9f0308adfcc0157fc29L17
It seems the middleware was removed and no longer needed.

**Environment**

* React-admin version: 2.3.0
* React version: 16.3.0

Also, it seems the import of `formReducer` is missing. I guessed it should have been:
```js
import { reducer as formReducer } from 'redux-form'
```

### PR DESCRIPTION
**Pull request by MohammedFaragallah**
Closes #2302

### CODE REVIEW

File: docs/CustomApp.md [hunk 1]
```diff
@@ -19,8 +19,10 @@ Here is the default store creation for react-admin:
 
 ```js
 // in src/createAdminStore.js
-import { combineReducers, createStore, compose, applyMiddleware } from 'redux';
-import { routerMiddleware } from 'react-router-redux';
+import { applyMiddleware, combineReducers, compose, createStore } from 'redux';
+import { routerMiddleware, routerReducer } from 'react-router-redux';
+import formMiddleware from 'ra-core/lib/form/formMiddleware';
```
Comment (fzaninotto) [inline docs/CustomApp.md]: There is indeed a problem, but you don't have the right fix. Importing modules from their path in the compiled code isn't supported in react-admin.

The problem is that the `formMiddelware` isn't exported by `ra-core`, and this should be fixed in `packages/ra-core/src/createAdminStore.ts`.

Comment (MohammedFaragallah) [reply docs/CustomApp.md]: Thanks for the clarification I'll try to make the needed changes ASAP. I'm not used to working with open source but I really like your project it helped me a lot.

**Comment by fzaninotto** [comment]
Any progress?

**Comment by fzaninotto** [comment]
Superseded by #2828

**Pull request closed without merging**